File: arrays/transposeMatrix.java
// https://leetcode.com/problems/transpose-matrix/
package arrays;

public class transposeMatrix {
    public static void main(String[] args) {
        SolutionTM solution = new SolutionTM() ;
        print(solution.transpose(new int[][]{{1,2,3},{4,5,6},{7,8,9}}));
        print(solution.transpose(new int[][]{{1,2,3},{4,5,6}}));   

    }

    public static void print(int[][] matrix){
        for (int[] i : matrix){
            for(int j : i){
                System.out.print(j + " ");
            }
            System.out.println();
        }
    }
}

class SolutionTM {
    public int[][] transpose(int[][] matrix) {

        int[][] result = new int[matrix[0].length][matrix.length];
        
        for(int i=0; i< matrix.length; i++){
            for (int j =0 ; j<matrix[i].length;j++){
                result[j][i]= matrix[i][j];
                
            }
        }
        return result;
        
    }
    
}

File: arrays/transposeMatrix.py
# https://leetcode.com/problems/transpose-matrix/

class Solution(object):
    
    # Runtime: 78 ms, faster than 42.44% of Python online submissions for Transpose Matrix.
    # Memory Usage: 14.2 MB, less than 86.05% of Python online submissions for Transpose Matrix.
    def transpose(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        
        
        result = [[0]*len(matrix) for i in range (len(matrix[0]))]
        
        for i in range (len(matrix)):
            for j in range (len(matrix[0])):
                result[j][i] = matrix[i][j]
                
        return result
    
    # Runtime: 52 ms, faster than 91.86% of Python online submissions for Transpose Matrix.
    # Memory Usage: 14.4 MB, less than 40.12% of Python online submissions for Transpose Matrix.
    def transpose(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        return list(zip(*matrix))